packages/@aws-cdk/aws-servicecatalog/test/portfolio.test.ts

```diff
@@ -0,0 +1,271 @@
+    const stack = new Stack(app);
+    const shareAccountId = '012345678901';
+
+    const portfolio = new Portfolio(stack, 'MyPortfolio', {
+      name: 'testPortfolio',
+      provider: 'testProvider',
+    });
+
+    portfolio.share(shareAccountId, {
+      shareTagOptions: true,
+      acceptLanguage: AcceptLanguage.EN,
+    });
+
+    expect(stack).toHaveResource('AWS::ServiceCatalog::PortfolioShare', {
+      AccountId: shareAccountId,
+      ShareTagOptions: true,
+      AcceptLanguage: 'en',
+    });
+  }),
+
+  test('portfolio share without share tagOptions', () => {
+    const app = new App();
+    const stack = new Stack(app);
+    const shareAccountId = '012345678901';
+
+    const portfolio = new Portfolio(stack, 'MyPortfolio', {
+      name: 'testPortfolio',
+      provider: 'testProvider',
+    });
+
+    portfolio.share(shareAccountId, { shareTagOptions: false });
+
+    expect(stack).toHaveResource('AWS::ServiceCatalog::PortfolioShare', {
+      AccountId: shareAccountId,
+      ShareTagOptions: false,
+    });
+  }),
+
+  test('portfolio share without explicit share tagOptions', () => {
+    const app = new App();
+    const stack = new Stack(app);
+    const shareAccountId = '012345678901';
+
+    const portfolio = new Portfolio(stack, 'MyPortfolio', {
+      name: 'testPortfolio',
+      provider: 'testProvider',
+    });
+
+    portfolio.share(shareAccountId);
+
+    expect(stack).toHaveResource('AWS::ServiceCatalog::PortfolioShare', {
+      AccountId: shareAccountId,
+    });
+  }),
+
+  test('portfolio principal association with role type', () => {
+    const app = new App();
+    const stack = new Stack(app);
+
+    const portfolio = new Portfolio(stack, 'MyPortfolio', {
+      name: 'testPortfolio',
+      provider: 'testProvider',
+    });
+
+    const role = new Role(stack, 'TestRole', {
+      assumedBy: new AccountRootPrincipal(),
+    });
+
+    portfolio.giveAccessToRole(role);
+
+    expect(stack).toHaveResourceLike('AWS::ServiceCatalog::PortfolioPrincipalAssociation', {});
+  }),
+
+  test('portfolio principal association with user type', () => {
+    const app = new App();
+    const stack = new Stack(app);
+
+    const portfolio = new Portfolio(stack, 'MyPortfolio', {
+      name: 'testPortfolio',
+      provider: 'testProvider',
+    });
+
+    const user = new User(stack, 'TestUser', {});
+
+    portfolio.giveAccessToUser(user);
+
+    expect(stack).toHaveResourceLike('AWS::ServiceCatalog::PortfolioPrincipalAssociation', {});
+  }),
+
+  test('portfolio principal association with group type', () => {
+    const app = new App();
+    const stack = new Stack(app);
+
+    const portfolio = new Portfolio(stack, 'MyPortfolio', {
+      name: 'testPortfolio',
+      provider: 'testProvider',
+    });
+
+    const group = new Group(stack, 'TestGroup', {});
+
+    portfolio.giveAccessToGroup(group);
+
+    expect(stack).toHaveResourceLike('AWS::ServiceCatalog::PortfolioPrincipalAssociation', {});
```

Comment:
No need for the empty object.

```suggestion
    expect(stack).toHaveResourceLike('AWS::ServiceCatalog::PortfolioPrincipalAssociation');
```

BTW, is there really no interesting property to assert here...?

Reply:
adjusted to add fields